src/CMakeLists.txt

```diff
@@ -5,25 +5,25 @@ set(RandLAPACK_cxx_srcs
     comps/rs.cc
     comps/util.cc
     comps/orth.cc
-    comps/preconditioners.cc
 )
 
 set(RandLAPACK_libs
-    RandLAPACK_RandBLAS
-    RandLAPACK_lapackpp
-    RandBLAS_blaspp
+    RandBLAS
+    lapackpp
+    blaspp
     RandBLAS_Random123
-    RandBLAS_OpenMP
+    OpenMP::OpenMP_CXX
 )
 
 add_library(RandLAPACK ${RandLAPACK_cxx_srcs})
+target_compile_options(RandLAPACK PUBLIC -Wall -Wextra)
 
 target_include_directories(
     RandLAPACK PUBLIC
-    $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/include>
-    $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/include/RandLAPACK>
-    $<INSTALL_INTERFACE:include>
-    $<INSTALL_INTERFACE:include/RandLAPACK>
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../include>
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../include/RandLAPACK>
```

Comment:
What files do we expect to be located in ``{CMAKE_CURRENT_SOURCE_DIR}/../``? Is this just robustness of the cmake build system?

Reply:
this is to support nesting projects. `CMAKE_SOURCE_DIR` is always the top level. When we are a git submodule it is the wrong path.